### PR TITLE
fix(float-equal-update): add "type: module", add test

### PR DESCRIPTION
**Pull request by hkleungai**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

**Comment by dmurvihill** [comment]
Good morning @hkleungai, thanks for submitting.

You might have to educate me on how ES modules works in the context of DefinitelyTyped. The [float-equal](https://github.com/sindresorhus/float-equal/blob/main/package.json) package is still on CJS, so my instinct is to leave this module on CJS as well. Can you summarize how this is going to impact imports?

**Comment by hkleungai** [comment]
Hi @dmurvihill, you may want to try run test on this package before this commit, and I think you shall see this error.

> \> definitely-typed@0.0.3 test /Users/jimmyleung/Documents/DefinitelyTyped
> \> node --enable-source-maps node_modules/@definitelytyped/dtslint/ types "float-equal"
> 
> dtslint@0.2.23
> Error: !@arethetypeswrong/cli
> float-equal v3.0.0
> @types/float-equal v3.0.9999
> 
>  (ignoring rules: 'no-resolution', 'cjs-only-exports-default', 'unexpected-module-syntax', 'cjs-resolves-to-esm')
> 
> 🎭 Import resolved to a CommonJS type declaration file, but an ESM JavaScript file. https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/FalseCJS.md

There are some points worth to note in the _Explanation_ section of the readme file.
- In Node, files are determined to be ES modules or CommonJS modules by their file extension and ancestor package.json `"type"` field. When in `--moduleResolution node16`, TypeScript uses the same algorithm to determine the module format of type declaration files.
- When a user writes an import, TypeScript needs to know whether the resolved module is ESM or CJS in order to provide accurate checking. It makes this determination based on the file extension and package.json "type" of the type declaration file it finds.
- This (FalseCJS) problem indicates a violation of that assumption where the type declaration file implies that the corresponding runtime module is CommonJS, but it appears that Node will resolve to an ES module.

So to answer your question, no, this module is no longer on CJS since v3. And this fact need to be passed to TS consumer via the newly-added `"type": "module"` in the package.json.

You may also check that, in https://github.com/sindresorhus/float-equal/commits/main/package.json, v3 is already in ESM.

**Comment by dmurvihill** [comment]
Wow... I think I need new glasses. Thanks!

**Review by dmurvihill — APPROVED** [review]

**Review by sandersn — APPROVED** [review]

**Comment by hkleungai** [comment]
Ready to merge
